### PR TITLE
[Confluence] Ingest full body of documents in sync rules

### PR DESCRIPTION
**Pull request by praveen-elastic**
## Closes https://github.com/elastic/connectors-py/issues/2582

Following changes have been made in this PR:
- Ingest full body of documents while using advanced sync rules.

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)
- [ ] Considered corresponding documentation changes
- [ ] Contributed any configuration settings changes to the configuration reference
- [ ] if you added or changed Rich Configurable Fields for a Native Connector, you made a corresponding PR in [Kibana](https://github.com/elastic/kibana/blob/main/packages/kbn-search-connectors/types/native_connectors.ts)

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

### CODE REVIEW

File: connectors/sources/confluence.py [hunk 1]
```diff
@@ -959,7 +959,7 @@ async def search_by_query(self, query):
                 "_id": entity_details.get("id"),
                 "title": entity.get("title"),
                 "_timestamp": entity.get("lastModified"),
-                "body": entity.get("excerpt"),
+                "body": entity_details.get("body", {}).get("storage", {}).get("value"),
```
Comment (artem-shelkovnikov) [inline connectors/sources/confluence.py:962]: Can we use the same safe function as [here](https://github.com/elastic/connectors/pull/2598)?

**Comment by praveen-elastic** [comment]
> Does it slow down the sync with advanced rules by chance?

No, since there won't be any extra api call, sync would take same time as before.

File: connectors/sources/confluence.py (hunk 1 above)
Comment (praveen-elastic) [reply connectors/sources/confluence.py:962]: Sure, i'll raise a follow-up PR and address this in all relevant places once [that PR](https://github.com/elastic/connectors/pull/2598) is merged.

**Comment by seanstory** [comment]
@praveen-elastic can you please manually backport? Looks like there are conflicts ☝️

**Pull request merged**